core/src/test/scala/com/nvidia/spark/rapids/tool/qualification/QualificationSuite.scala

```diff
@@ -1373,6 +1373,58 @@ class QualificationSuite extends BaseTestSuite {
     }
   }
 
+  test("test SortMergeJoin corner case not supported on left join") {
+    TrampolineUtil.withTempDir { outParquetFile =>
+      TrampolineUtil.withTempDir { eventLogDir =>
+        val (eventLog, _) = ToolTestUtils.generateEventLog(eventLogDir, "sortMergeJoin") { spark =>
+          import spark.implicits._
+          val data = Seq(("A", 20, "M", "2024-01-01"),
+            ("B", 25, "M", "2024-12-12"), ("C", 30, "F", "2022-03-04"))
+              .toDF("name", "id", "gender", "dob_str")
+
+          data.write.format("parquet").mode("overwrite").save(outParquetFile.getCanonicalPath)
+          spark.read.parquet(outParquetFile.getCanonicalPath).createOrReplaceTempView("df")
+          spark.conf.set("spark.rapids.sql.hasExtendedYearValues","false")
+          spark.conf.set("spark.sql.autoBroadcastJoinThreshold","1")
+          val res = spark.sql("SELECT COUNT(*) FROM df a LEFT JOIN df b ON " +
+              "lower(a.id) in ('1','2') AND a.name = b.name")
+          res
+        }
+        val reader = FSUtils.readFileContentAsUTF8(eventLog)
+        assert(reader.contains("SortMergeJoin"))
```

Comment:
Could this condition be true always since app name would have the keyword `SortMergeJoin`?

Reply:
Thanks @parthosa. Good catch. Updated the test and appName. PTAL.